src/client/auth/oidc.rs

```diff
@@ -246,14 +374,15 @@ pub(crate) async fn build_client_first(
     credential: &Credential,
     server_api: Option<&ServerApi>,
 ) -> Option<Command> {
-    credential.oidc_callback.as_ref()?;
+    credential.oidc_callback.inner.lock().await.as_ref()?;
```

Comment:
I think this can be simpler with something like:
```rust
let guard = credential.oidc.callback.inner.lock().await;
let access_token = guard.as_ref()?.cache.access_token.as_ref()?;
...
```

Reply:
I think this was just some cruft from my initial implementation a few weeks ago. I think it's enough just to replace the unwrap in the if with `?`